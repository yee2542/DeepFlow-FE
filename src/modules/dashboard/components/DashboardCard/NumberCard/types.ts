export type NumberCardProps = {
  cardName: string;
  largeNumber: number;
  todayNumber: number;
};
