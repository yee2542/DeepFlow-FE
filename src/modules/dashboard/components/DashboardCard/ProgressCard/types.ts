export type ProgressCardProps = {
  cardName: string;
  progress: number;
  doneNumber: number;
  totalNumber: number;
};
