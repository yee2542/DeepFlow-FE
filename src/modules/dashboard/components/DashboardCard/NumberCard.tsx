import React from 'react';
import kFormatter from 'Utils/kFormatter';
import { DashboardCard, SmallBodyText } from './styled';
import { NumberCardProps } from './types';

const NumberCard: React.FC<NumberCardProps> = (props) => {
  const { cardName, largeNumber, todayNumber } = props;
  return (
    <DashboardCard>
      <h5>{cardName}</h5>
      <br />
      <h3>{kFormatter(largeNumber)}</h3>
      <SmallBodyText>today upload</SmallBodyText>
      <p style={{ fontWeight: 'bold', color: '#5A36CC' }}>+{kFormatter(todayNumber)}</p>
    </DashboardCard>
  );
};

export default NumberCard;
