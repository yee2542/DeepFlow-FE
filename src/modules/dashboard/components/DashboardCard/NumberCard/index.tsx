import React from 'react';
import kFormatter from 'Utils/kFormatter';
import { DashboardCard, SmallBodyText, TodayNumberText } from '../styled';
import { NumberCardProps } from './types';

const NumberCard: React.FC<NumberCardProps> = (props) => {
  const { cardName, largeNumber, todayNumber } = props;
  return (
    <DashboardCard>
      <h5>{cardName}</h5>
      <br />
      <h3>{kFormatter(largeNumber)}</h3>
      <SmallBodyText>today upload</SmallBodyText>
      <TodayNumberText>+{kFormatter(todayNumber)}</TodayNumberText>
    </DashboardCard>
  );
};

export default NumberCard;
