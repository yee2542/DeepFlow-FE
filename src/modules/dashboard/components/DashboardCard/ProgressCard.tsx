import { Progress } from 'antd';
import React from 'react';
import { DashboardCard, SmallBodyText } from './styled';
import { _progressCardProps } from './types';

const ProgressCard: React.FC<_progressCardProps> = (props) => {
  const { cardName, progress, doneNumber, totalNumber } = props;
  return (
    <DashboardCard>
      <h5>{cardName}</h5>
      <div style={{ textAlign: 'center' }}>
        <Progress
          type="circle"
          percent={progress}
          strokeWidth={8}
          strokeColor={{ '0%': '#5a36cc', '100%': '#5a36cc' }}
        />
        <SmallBodyText style={{ marginTop: '10%' }}>
          {doneNumber} of {totalNumber} photos
        </SmallBodyText>
      </div>
    </DashboardCard>
  );
};

export default ProgressCard;
