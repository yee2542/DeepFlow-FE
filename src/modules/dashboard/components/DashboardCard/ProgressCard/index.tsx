import { Progress } from 'antd';
import React from 'react';
import {
  DashboardCard,
  SmallBodyText,
  ProgressCardWrapper,
} from 'Modules/dashboard/components/DashboardCard/styled';
import { ProgressCardProps } from './types';
import kFormatter from 'Utils/kFormatter';

const ProgressCard: React.FC<ProgressCardProps> = (props) => {
  const { cardName, progress, doneNumber, totalNumber } = props;
  return (
    <DashboardCard>
      <h5>{cardName}</h5>
      <ProgressCardWrapper>
        <Progress
          type="circle"
          percent={progress}
          strokeWidth={8}
          strokeColor={{ '0%': '#5a36cc', '100%': '#5a36cc' }}
        />
        <SmallBodyText style={{ marginTop: '10%' }}>
          {kFormatter(doneNumber)} of {kFormatter(totalNumber)} photos
        </SmallBodyText>
      </ProgressCardWrapper>
    </DashboardCard>
  );
};

export default ProgressCard;
