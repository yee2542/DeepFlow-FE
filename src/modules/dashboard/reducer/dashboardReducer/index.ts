import { createSlice } from '@reduxjs/toolkit';
import { initDashboardState } from './init';
import { DASHBOARD } from '../../models/types';
import { fetchDashboardData } from './actions';
import { ErrorStateCodeEnum } from 'Stores/common/types/error';

export const dashboardSlice = createSlice({
  name: DASHBOARD,
  initialState: initDashboardState,
  reducers: {
    init(state) {
      return { ...state, ...initDashboardState };
    },
  },
  extraReducers: (builder) => {
    builder.addCase(fetchDashboardData.rejected, (state) => {
      state.ready = false;
      state = {
        ...state,
        error: {
          code: ErrorStateCodeEnum.ServerInternalError,
          msg: 'api error',
        },
      };
    });
    builder.addCase(fetchDashboardData.pending, (state) => {
      state.ready = false;
      state.error = undefined;
    });
    builder.addCase(fetchDashboardData.fulfilled, (state, action) => {
      state.ready = true;
      state.error = undefined;
      state.data.albumPreviewData = action.payload.data.AlbumPreviewMockData;
      state.data.dashboardData = action.payload.data.DashboardMockData;
    });
  },
});

export default dashboardSlice.reducer;
export const dashboardActions = dashboardSlice.actions;
