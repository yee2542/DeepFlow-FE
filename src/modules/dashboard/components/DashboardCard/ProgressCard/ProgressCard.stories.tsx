import { Meta, Story } from '@storybook/react';
import ProgressCard from 'Modules/dashboard/components/DashboardCard/ProgressCard/index';
import { ProgressCardProps } from './types';
import React from 'react';
import { GlobalStyle } from 'Styles/global';
import { Row, Col } from 'antd';
import { DashboardWrapper, CardWrapper } from 'Modules/dashboard/pages/styled';

export default {
  title: 'FluentSearch/DashboardCard/ProgressCard',
  component: ProgressCard,
} as Meta;

const ProgressCardStory: Story<ProgressCardProps> = (args) => (
  <>
    <GlobalStyle />
    <DashboardWrapper>
      <CardWrapper>
        <Row justify="start">
          <Col md={4}>
            <ProgressCard {...args} />
          </Col>
        </Row>
      </CardWrapper>
    </DashboardWrapper>
  </>
);

export const Default = ProgressCardStory.bind({});
Default.args = {
  cardName: 'cardName',
  progress: 75,
  doneNumber: 125300,
  totalNumber: 125300,
};
