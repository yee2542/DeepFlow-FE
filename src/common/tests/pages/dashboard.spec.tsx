import { mount } from 'enzyme';
import React from 'react';
import DashboardPage from 'Modules/dashboard/pages';
import NumberCard from 'Modules/dashboard/components/DashboardCard/NumberCard/index';
import ProgressCard from 'Modules/dashboard/components/DashboardCard/ProgressCard/index';
import dashboardReducer from 'Modules/dashboard/reducer/dashboardReducer';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

const rootReducer = combineReducers({
  dashboard: dashboardReducer,
});

const store = configureStore({
  reducer: rootReducer,
});

describe('Pages/Dashboard test', () => {
  it('Dashboard should be exisiting', () => {
    const wrap = mount(
      <Provider store={store}>
        <DashboardPage />
      </Provider>,
    );
    expect(wrap.exists()).toBe(true);
  });

  it('Should have Overview Album', () => {
    const wrap = mount(
      <Provider store={store}>
        <DashboardPage />
      </Provider>,
    );
    expect(wrap.exists('OverviewAlbum')).toBe(true);
  });

  it('Should contain 2 NumberCard, 1 Progress Card and 1 Model Card', () => {
    const wrap = mount(
      <Provider store={store}>
        <DashboardPage />
      </Provider>,
    );
    expect(wrap.find(NumberCard)).toHaveLength(2);
    expect(wrap.find(ProgressCard)).toHaveLength(1);
    expect(wrap.find('ModelCard')).toHaveLength(1);
  });
});
