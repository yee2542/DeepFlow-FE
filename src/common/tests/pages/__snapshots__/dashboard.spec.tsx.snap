// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Dashboard test Render correctly 1`] = `
"<div class=\\"styled__DashboardWrapper-o6ajqd-0 kGoakK\\"><div class=\\"styled__ContentWrapper-o6ajqd-2 jSXSpD\\"><h3 class=\\"styled__DashboardHeader-o6ajqd-5 byGPKm\\">Dashboard</h3><div class=\\"styled__CardWrapper-o6ajqd-3 hgUsVP\\"><div class=\\"ant-row ant-row-space-around ant-row-middle\\"><div class=\\"ant-col ant-col-md-5\\"><div class=\\"ant-card ant-card-bordered styled__DashboardCard-yz824c-0 gPxDtN\\"><div class=\\"ant-card-body\\"><h5>Photos</h5><br/><h3>123.0k</h3><p class=\\"styled__SmallBodyText-yz824c-1 eQfubR\\">today upload</p><p class=\\"styled__TodayNumberText-yz824c-3 bGdmGW\\">+123.0k</p></div></div></div><div class=\\"ant-col ant-col-md-5\\"><div class=\\"ant-card ant-card-bordered styled__DashboardCard-yz824c-0 gPxDtN\\"><div class=\\"ant-card-body\\"><h5>Videos</h5><br/><h3>123.0k</h3><p class=\\"styled__SmallBodyText-yz824c-1 eQfubR\\">today upload</p><p class=\\"styled__TodayNumberText-yz824c-3 bGdmGW\\">+123.0k</p></div></div></div><div class=\\"ant-col ant-col-md-5\\"><div class=\\"ant-card ant-card-bordered styled__DashboardCard-yz824c-0 gPxDtN\\"><div class=\\"ant-card-body\\"><h5>Task</h5><div class=\\"styled__ProgressCardWrapper-yz824c-4 bfHiKh\\"><div class=\\"ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default\\"><div class=\\"ant-progress-inner ant-progress-circle-gradient\\" style=\\"width:120px;height:120px;font-size:24px\\"><svg class=\\"ant-progress-circle\\" viewBox=\\"0 0 100 100\\"><defs><linearGradient id=\\"ant-progress-gradient-1\\" x1=\\"100%\\" y1=\\"0%\\" x2=\\"0%\\" y2=\\"0%\\"><stop offset=\\"0%\\" stop-color=\\"#5a36cc\\"></stop><stop offset=\\"100%\\" stop-color=\\"#5a36cc\\"></stop></linearGradient></defs><path class=\\"ant-progress-circle-trail\\" d=\\"M 50,50 m 0,-46
   a 46,46 0 1 1 0,92
   a 46,46 0 1 1 0,-92\\" stroke-linecap=\\"round\\" stroke-width=\\"8\\" fill-opacity=\\"0\\" style=\\"stroke-dasharray:289.02652413026095px 289.02652413026095px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s\\"></path><path class=\\"ant-progress-circle-path\\" d=\\"M 50,50 m 0,-46
   a 46,46 0 1 1 0,92
   a 46,46 0 1 1 0,-92\\" stroke=\\"url(#ant-progress-gradient-1)\\" stroke-linecap=\\"round\\" stroke-width=\\"8\\" opacity=\\"1\\" fill-opacity=\\"0\\" style=\\"stroke:[object Object];stroke-dasharray:216.76989309769573px 289.02652413026095px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s\\"></path></svg><span class=\\"ant-progress-text\\" title=\\"75%\\">75%</span></div></div><p style=\\"margin-top:10%\\" class=\\"styled__SmallBodyText-yz824c-1 eQfubR\\">100 of 1.0k photos</p></div></div></div></div><div class=\\"ant-col ant-col-md-5\\"><div class=\\"ant-card ant-card-bordered styled__DashboardCard-yz824c-0 gPxDtN\\"><div class=\\"ant-card-body\\"><h5>Model</h5><br/><span class=\\"ant-tag ant-tag-purple styled__Label-yz824c-2 bKWFRV\\" style=\\"margin-top:10%\\">resNet</span><h3 style=\\"margin-top:0%;margin-bottom:-3%\\">800</h3><p>photos</p></div></div></div></div></div><div class=\\"styled__AlbumWrapper-o6ajqd-4 hirseO\\"><h3>Lastest Photo</h3><div class=\\"ant-row ant-row-space-around ant-row-middle\\"><div class=\\"ant-col ant-col-md-5\\"><div class=\\"styled__AlbumWrapper-koalmt-0 chvFDA\\"><div class=\\"styled__Container-koalmt-2 UPZmO\\"><div style=\\"display:inline-block;max-width:100%;overflow:hidden;position:relative;box-sizing:border-box;margin:0\\"><div style=\\"box-sizing:border-box;display:block;max-width:100%\\"><img style=\\"max-width:100%;display:block\\" alt=\\"\\" aria-hidden=\\"true\\" role=\\"presentation\\" src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4=\\"/></div><img src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" style=\\"visibility:hidden;position:absolute;top:0;left:0;bottom:0;right:0;box-sizing:border-box;padding:0;border:none;margin:auto;display:block;width:0;height:0;min-width:100%;max-width:100%;min-height:100%;max-height:100%\\"/></div><div id=\\"text\\" class=\\"styled__Text-koalmt-3 fjaVqR\\">View Photo</div></div><span><b>album name</b></span><br/><span>1 items</span><br/><div style=\\"margin-top:10px\\"><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label1</span><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label2</span><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label3</span></div></div></div><div class=\\"ant-col ant-col-md-5\\"><div class=\\"styled__AlbumWrapper-koalmt-0 chvFDA\\"><div class=\\"styled__Container-koalmt-2 UPZmO\\"><div style=\\"display:inline-block;max-width:100%;overflow:hidden;position:relative;box-sizing:border-box;margin:0\\"><div style=\\"box-sizing:border-box;display:block;max-width:100%\\"><img style=\\"max-width:100%;display:block\\" alt=\\"\\" aria-hidden=\\"true\\" role=\\"presentation\\" src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4=\\"/></div><img src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" style=\\"visibility:hidden;position:absolute;top:0;left:0;bottom:0;right:0;box-sizing:border-box;padding:0;border:none;margin:auto;display:block;width:0;height:0;min-width:100%;max-width:100%;min-height:100%;max-height:100%\\"/></div><div id=\\"text\\" class=\\"styled__Text-koalmt-3 fjaVqR\\">View Photo</div></div><span><b>album name</b></span><br/><span>1 items</span><br/><div style=\\"margin-top:10px\\"><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label1</span><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label2</span><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label3</span></div></div></div><div class=\\"ant-col ant-col-md-5\\"><div class=\\"styled__AlbumWrapper-koalmt-0 chvFDA\\"><div class=\\"styled__Container-koalmt-2 UPZmO\\"><div style=\\"display:inline-block;max-width:100%;overflow:hidden;position:relative;box-sizing:border-box;margin:0\\"><div style=\\"box-sizing:border-box;display:block;max-width:100%\\"><img style=\\"max-width:100%;display:block\\" alt=\\"\\" aria-hidden=\\"true\\" role=\\"presentation\\" src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4=\\"/></div><img src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" style=\\"visibility:hidden;position:absolute;top:0;left:0;bottom:0;right:0;box-sizing:border-box;padding:0;border:none;margin:auto;display:block;width:0;height:0;min-width:100%;max-width:100%;min-height:100%;max-height:100%\\"/></div><div id=\\"text\\" class=\\"styled__Text-koalmt-3 fjaVqR\\">View Photo</div></div><span><b>album name</b></span><br/><span>1 items</span><br/><div style=\\"margin-top:10px\\"><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label1</span><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label2</span><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label3</span></div></div></div><div class=\\"ant-col ant-col-md-5\\"><div class=\\"styled__AlbumWrapper-koalmt-0 chvFDA\\"><div class=\\"styled__Container-koalmt-2 UPZmO\\"><div style=\\"display:inline-block;max-width:100%;overflow:hidden;position:relative;box-sizing:border-box;margin:0\\"><div style=\\"box-sizing:border-box;display:block;max-width:100%\\"><img style=\\"max-width:100%;display:block\\" alt=\\"\\" aria-hidden=\\"true\\" role=\\"presentation\\" src=\\"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4=\\"/></div><img src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" decoding=\\"async\\" style=\\"visibility:hidden;position:absolute;top:0;left:0;bottom:0;right:0;box-sizing:border-box;padding:0;border:none;margin:auto;display:block;width:0;height:0;min-width:100%;max-width:100%;min-height:100%;max-height:100%\\"/></div><div id=\\"text\\" class=\\"styled__Text-koalmt-3 fjaVqR\\">View Photo</div></div><span><b>album name</b></span><br/><span>1 items</span><br/><div style=\\"margin-top:10px\\"><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label1</span><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label2</span><span class=\\"ant-tag ant-tag-purple styled__Label-koalmt-1 kNyTJl\\">label3</span></div></div></div></div></div></div></div>"
`;
